In [1]:
import pandas as pd

# Read CSV Files
A simple way to store big data sets is to use CSV files (comma separated files).

CSV files contains plain text and is a well know format that can be read by everyone including Pandas.

In our examples we will be using a CSV file called 'gym.csv'.

In [3]:
df = pd.read_csv("gym.csv")

In [4]:
print(df)

          Date  workout     time
0   31-01-2025     walk  1:00:00
1   01-02-2025     walk  1:00:00
2   02-02-2025     walk  1:00:00
3   03-02-2025     Work        0
4   04-02-2025    Slept        0
5   05-02-2025  Holiday        0
6   06-02-2025      Gym  0:55:00
7   06-02-2025     Walk  0:45:00
8   07-02-2025      Gym  0:50:00
9   07-02-2025     Walk  0:20:00
10  08-02-2025     Work        0
11  09-02-2025     Walk  0:45:00
12  10-02-2025     Work        0
13  11-02-2025      Gym  0:45:00
14  11-02-2025     Walk  0:20:00
15  12-02-2025     Walk  1:15:00
16  13-02-2025  Holiday        0
17  14-02-2025     Walk  0:45:00
18  15-02-2025     Walk  1:00:00
19  16-02-2025     Walk  1:00:00
20  17-02-2025     Work        0
21  18-02-2025     Walk  0:50:00
22  19-02-2025  Holiday        0
23  20-02-2025  Holiday        0
24  21-02-2025     Walk  0:45:00
25  22-02-2025     Walk  1:00:00
26  23-02-2025     Walk  1:00:00
27  24-02-2025     Work        0
28  25-02-2025     Walk  1:00:00
29  26-02-

In [5]:
print(df.to_string())

          Date  workout     time
0   31-01-2025     walk  1:00:00
1   01-02-2025     walk  1:00:00
2   02-02-2025     walk  1:00:00
3   03-02-2025     Work        0
4   04-02-2025    Slept        0
5   05-02-2025  Holiday        0
6   06-02-2025      Gym  0:55:00
7   06-02-2025     Walk  0:45:00
8   07-02-2025      Gym  0:50:00
9   07-02-2025     Walk  0:20:00
10  08-02-2025     Work        0
11  09-02-2025     Walk  0:45:00
12  10-02-2025     Work        0
13  11-02-2025      Gym  0:45:00
14  11-02-2025     Walk  0:20:00
15  12-02-2025     Walk  1:15:00
16  13-02-2025  Holiday        0
17  14-02-2025     Walk  0:45:00
18  15-02-2025     Walk  1:00:00
19  16-02-2025     Walk  1:00:00
20  17-02-2025     Work        0
21  18-02-2025     Walk  0:50:00
22  19-02-2025  Holiday        0
23  20-02-2025  Holiday        0
24  21-02-2025     Walk  0:45:00
25  22-02-2025     Walk  1:00:00
26  23-02-2025     Walk  1:00:00
27  24-02-2025     Work        0
28  25-02-2025     Walk  1:00:00
29  26-02-

# max_rows
The number of rows returned is defined in Pandas option settings.

You can check your system's maximum rows with the pd.options.display.max_rows statement.

In [6]:
print(pd.options.display.max_rows)

60


In my system the number is 60, which means that if the DataFrame contains more than 60 rows, the print(df) statement will return only the headers and the first and last 5 rows.

You can change the maximum rows number with the same statement.

In [10]:
pd.set_option("display.max_rows", 100)

# Read JSON
Big data sets are often stored, or extracted as JSON.

JSON is plain text, but has the format of an object, and is well known in the world of programming, including Pandas.

In our examples we will be using a JSON file called 'data.json'.

In [11]:
df = pd.read_json('data.js')

In [12]:
print(df)

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.4
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[169 rows x 4 columns]


# Viewing the Data
One of the most used method for getting a quick overview of the DataFrame, is the head() method.

The head() method returns the headers and a specified number of rows, starting from the top.

There is also a tail() method for viewing the last rows of the DataFrame.

The tail() method returns the headers and a specified number of rows, starting from the bottom.

In [13]:
print(df.head())

   Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0


In [14]:
print(df.head(10))

   Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0
5        60    102       127     300.5
6        60    110       136     374.0
7        45    104       134     253.3
8        30    109       133     195.1
9        60     98       124     269.0


In [15]:
print(df.tail())

     Duration  Pulse  Maxpulse  Calories
164        60    105       140     290.8
165        60    110       145     300.4
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4


In [16]:
print(df.tail(10))

     Duration  Pulse  Maxpulse  Calories
159        30     80       120     240.9
160        30     85       120     250.4
161        45     90       130     260.4
162        45     95       130     270.0
163        45    100       140     280.9
164        60    105       140     290.8
165        60    110       145     300.4
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4


# Info About the Data
The DataFrames object has a method called info(), that gives you more information about the data set.

In [17]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
Index: 169 entries, 0 to 168
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Duration  169 non-null    int64  
 1   Pulse     169 non-null    int64  
 2   Maxpulse  169 non-null    int64  
 3   Calories  164 non-null    float64
dtypes: float64(1), int64(3)
memory usage: 6.6 KB
None


# Result Explained
The result tells us there are 169 rows and 4 columns:


  RangeIndex: 169 entries, 0 to 168
  Data columns (total 4 columns):

And the name of each column, with the data type:


   #   Column    Non-Null Count  Dtype  
  ---  ------    --------------  -----  
   0   Duration  169 non-null    int64  
   1   Pulse     169 non-null    int64  
   2   Maxpulse  169 non-null    int64  
   3   Calories  164 non-null    float64

# Null Values
The info() method also tells us how many Non-Null values there are present in each column, and in our data set it seems like there are 164 of 169 Non-Null values in the "Calories" column.

Which means that there are 5 rows with no value at all, in the "Calories" column, for whatever reason.

Empty values, or Null values, can be bad when analyzing data, and you should consider removing rows with empty values. This is a step towards what is called cleaning data.